commitizen/git.py

```diff
@@ -164,3 +187,31 @@ def is_git_project() -> bool:
     if c.out.strip() == "true":
         return True
     return False
+
+
+def get_eol_style() -> EOLTypes:
+    c = cmd.run("git config core.eol")
```

Comment:
Can you improve this part of code running the git command only at the first call of `smart_open`, then saving `eol` for future `smart_open` calls ?

Reply:
Yep, it could be an improvement. But might not be an urgent one. I don't think it would be a huge overhead

Reply:
It may be a little hard – every repositories have their different configuration, and cache the configuration with their repository hash may be slower than run this command directly:

```
$ hyperfine "git config core.eol" -i -N -w 1000 -r 5000

Benchmark 1: git config core.eol
  Time (mean ± σ):       3.0 ms ±   0.4 ms    [User: 1.4 ms, System: 0.7 ms]
  Range (min … max):     2.6 ms …  10.5 ms    5000 runs
 
  Warning: Ignoring non-zero exit code.
  Warning: Statistical outliers were detected. Consider re-running this benchmark on a quiet PC without any interferences from other programs. It might help to use the '--warmup' or '--prepare' options.
```

It took about 700us to run `git config core.eol`, which may not be a noticeable overhead when looking into other Git commands that will be executed.